### PR TITLE
Fix warnings while running tox -e doc command.

### LINKED ISSUE #887: Issues when building the docs
When building the docs (`tox -e docs`) we get some WARNINGs. It would be nice if some could take care of them.

```
dateparser/dateparser/date.py:docstring of dateparser.date.DateDataParser.get_date_data:34: WARNING: Unknown target name: "settings".
WARNING: missing attribute mentioned in :members: option: module dateparser.calendars.jalali, attribute JalaliParser
WARNING: autodoc: failed to import module 'detection' from module 'dateparser.languages'; the following exception was raised:
No module named 'dateparser.languages.detection'
WARNING: autodoc: failed to import module 'language' from module 'dateparser.languages'; the following exception was raised:
No module named 'dateparser.languages.language'
/home/marc/Projects/dateparser/dateparser/__init__.py:docstring of dateparser:1: WARNING: duplicate object description of dateparser, other instance in dateparser, use :noindex: for one of them
/home/marc/Projects/dateparser/dateparser/__init__.py:docstring of dateparser.parse:1: WARNING: duplicate object description of dateparser.parse, other instance in dateparser, use :noindex: for one of them
/home/marc/Projects/dateparser/dateparser/__init__.py:docstring of dateparser:1: WARNING: duplicate object description of dateparser, other instance in index, use :noindex: for one of them
/home/marc/Projects/dateparser/dateparser/__init__.py:docstring of dateparser.parse:1: WARNING: duplicate object description of dateparser.parse, other instance in index, use :noindex: for one of them
/home/marc/Projects/dateparser/dateparser/search/__init__.py:docstring of dateparser.search:1: WARNING: duplicate object description of dateparser.search, other instance in index, use :noindex: for one of them
/home/marc/Projects/dateparser/dateparser/search/__init__.py:docstring of dateparser.search.search_dates:1: WARNING: duplicate object description of dateparser.search.search_dates, other instance in index, use :noindex: for one of them
looking for now-outdated files... none found
pickling environment... done
checking consistency... /home/marc/Projects/dateparser/docs/modules.rst: WARNING: document isn't included in any toctree
/home/marc/Projects/dateparser/docs/template.rst: WARNING: document isn't included in any toctree

```

### PR DESCRIPTION
**Pull request by kishan3**
Helps fix #887

### CODE REVIEW

**Review by Gallaecio — APPROVED** [review]

**Comment by kishan3** [comment]
Hi @noviluni I have fixed the remaining warnings too now.

File: docs/template.rst [hunk 1]
```diff
@@ -1,4 +1,4 @@
-.. _language-data-template:
+ .. _language-data-template:
```
Comment (Gallaecio) [inline docs/template.rst:1]:
```suggestion
.. _language-data-template:
```

Comment (kishan3) [reply docs/template.rst]: My bad :facepalm:

**Review by Gallaecio — APPROVED** [review]
Nice cleanup.

A good next step would be to make the documentation build fail in case of a warning. The `-W` parameter does that. But that can be done later in a separate pull request, I think.

**Review by noviluni — APPROVED** [review]
great! :rocket:  thanks @kishan3 ! :smile:

**Pull request merged**